lib/screens/v2/widget_instance/dup_alert.ex

```diff
@@ -81,145 +81,159 @@ defmodule Screens.V2.WidgetInstance.DupAlert do
 
-    if effect == :delay and delay_severity < 5 do
+    if delay_severity >= 5 do
+      # All delays >= 20 minutes get a partial alert on all 3 rotations
+      :partial
+    else
       # We don't show delays < 20 minutes
-      log_layout_mismatch(t, parameters, delay_severity)
+      log_layout_mismatch(t, delay_severity)
       :no_render
-    else
-      case do_alert_layout(parameters) do
-        :no_render ->
-          log_layout_mismatch(t, parameters, delay_severity)
-          :no_render
+    end
+  end
 
-        layout ->
-          layout
-      end
+  defp non_delay_alert_layout(t) do
+    parameters = get_layout_parameters(t)
+
+    case {t.rotation_index, Map.fetch(@parameters_to_layout_zero, parameters)} do
+      {:zero, {:ok, layout_zero}} ->
+        # The first page's layout maps directly from `parameters`.
+        layout_zero
+
+      {:one, {:ok, _layout_zero}} ->
+        # The second page always uses full-screen layout, as long as the alert is relevant.
+        :full_screen
+
+      {:two, {:ok, layout_zero}} ->
+        # The third page's layout depends on whether or not the screen has secondary departures configured.
+        #
+        # If the screen has secondary departures configured, it always gets a partial alert.
+        # If the screen does not have secondary departures configured, it gets the same layout as the first rotation.
+        #
+        # This is because we do not want to hide secondary departures (e.g. bus, CR) if they exist.
+        has_secondary_departures = t.screen.app_params.secondary_departures.sections != []
+
+        if has_secondary_departures,
+          do: :partial,
+          else: layout_zero
+
+      {_rotation_index, :error} ->
+        # We don't know what to do under these conditions.
+        # Log it, and prevent the alert from appearing.
+        log_layout_mismatch(t, :not_applicable)
+        :no_render
     end
   end
 
   # The widget _should_ always be valid given the constraints of what alerts
   # we generate DUP alert widgets for, but in case one slips through, we
   # should know about it!
-  defp log_layout_mismatch(t, parameters, delay_severity) do
-    {effect, location, affected_line_count, total_section_count, rotation_index} = parameters
-
-    labeled_parameters = [
-      effect: effect,
-      delay_severity: delay_severity,
-      location: location,
-      affected_line_count: affected_line_count,
-      total_section_count: total_section_count,
-      rotation_index: rotation_index
-    ]
-
-    identifiers = [alert_id: t.alert.id, stop_ids: Enum.join(t.screen.alerts.stop_ids, ",")]
-    log_fields = identifiers ++ labeled_parameters
-
-    Logger.info(
+  defp log_layout_mismatch(t, delay_severity) do
+    log_fields =
+      t
+      |> get_layout_parameters()
+      |> Map.merge(%{
+        delay_severity: delay_severity,
+        rotation_index: t.rotation_index,
+        alert_id: t.alert.id,
+        stop_id: t.screen.app_params.alerts.stop_id
```

Comment:
One note on this change: my previous code was wrong. It didn't access the field correctly, and was based on an older version of the config schema that allowed multiple stop IDs.